executor/src/witgen/vm_processor.rs

```diff
@@ -118,10 +125,28 @@ impl<'a, T: FieldElement> VmProcessor<'a, T> {
             if looping_period.is_none() && row_index != rows_left - 1 {
                 self.ensure_has_next_row(row_index);
                 self.compute_row(row_index, mutable_state);
+
+                if let Some(latch) = self.latch.as_ref() {
+                    // Evaluate latch expression and return if it evaluates to 1.
+                    let row_pair = RowPair::from_single_row(
+                        self.row(row_index),
+                        row_index + self.row_offset,
+                        self.fixed_data,
+                        UnknownStrategy::Unknown,
+                    );
+                    let latch_value = row_pair.evaluate(latch).unwrap().constant_value().unwrap();
```

Comment:
Why do you know that the value is constant and the evaluation works?

Reply:
Hmm, good point.

If the latch is unknown, it means the machine hasn't finished. I changed it so that `VmProcessor::run()` returns an `EvalValue` which is incomplete in that case. Then `Generator` asserts that it must complete.

(Note that this code path is never executed right now, because the main VM has a latch of `None`. I guess for secondary VMs, it would just propagate the incomplete `EvalValue` back to the caller, who might be able to fix it by providing more inputs.)